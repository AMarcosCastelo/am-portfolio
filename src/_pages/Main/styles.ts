import styled from 'styled-components'

export const Wrapper = styled.main`
  background: ${({ theme }) => theme.body};
  width: 100vw;
  height: 100vh;
  overflow: hidden;
  position: relative;

  h2,
  h3,
  h4,
  h5,
  h6 {
    font-family: 'Karla', sans-serif;
    font-weight: 500;
  }
`

export const Container = styled.div`
  padding: 2rem;
`
