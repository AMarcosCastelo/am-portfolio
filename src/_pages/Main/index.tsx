import { useState } from 'react'
import NextLink from 'next/link'
import { motion } from 'framer-motion'
import PowerButton from 'components/PowerButton'
import Logo from 'components/Logo'
import SocialIcons from 'components/SocialIcons'
import { YinYang } from 'components/AllSvgs'
import Intro from 'components/Intro'

import * as S from './styles'

const Main = () => {
  const [click, setClick] = useState(false)

  const handleClick = () => {
    setClick(!click)
  }

  return (
    <S.Wrapper>
      <S.Container>
        <SocialIcons theme={click ? 'dark' : 'light'} />
        <PowerButton />
        <Logo theme={click ? 'dark' : 'light'} />

        <S.DarkDiv click={click} />

        <S.Center click={click}>
          <YinYang
            onClick={handleClick}
            width={click ? 120 : 200}
            height={click ? 120 : 200}
          />
          <span>click here</span>
        </S.Center>

        <NextLink href="mailto:marcos-12marcos@hotmail.com" passHref>
          <S.Contact target="_blank">
            <motion.h2 whileHover={{ scale: 1.1 }} whileTap={{ scale: 0.9 }}>
              Say hi...
            </motion.h2>
          </S.Contact>
        </NextLink>

        <NextLink href="/work" passHref>
          <S.Work click={click}>
            <motion.h2 whileHover={{ scale: 1.1 }} whileTap={{ scale: 0.9 }}>
              Work
            </motion.h2>
          </S.Work>
        </NextLink>

        <S.BottomBar>
          <NextLink href="/about" passHref>
            <S.About click={click}>
              <motion.h2 whileHover={{ scale: 1.1 }} whileTap={{ scale: 0.9 }}>
                About
              </motion.h2>
            </S.About>
          </NextLink>

          <NextLink href="/mySkills" passHref>
            <S.MySkills>
              <motion.h2 whileHover={{ scale: 1.1 }} whileTap={{ scale: 0.9 }}>
                My Skills
              </motion.h2>
            </S.MySkills>
          </NextLink>
        </S.BottomBar>
      </S.Container>

      {click && <Intro />}
    </S.Wrapper>
  )
}

export default Main
