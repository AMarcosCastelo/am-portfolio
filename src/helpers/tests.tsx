import { ThemeProvider } from 'styled-components'
import { render, RenderResult } from '@testing-library/react'

import { LIGHT_THEME, DARK_THEME } from 'styles'

export const renderWithTheme = (
  children: React.ReactNode,
  theme?: 'light' | 'dark'
): RenderResult =>
  render(
    <ThemeProvider theme={theme === 'dark' ? DARK_THEME : LIGHT_THEME}>
      {children}
    </ThemeProvider>
  )
