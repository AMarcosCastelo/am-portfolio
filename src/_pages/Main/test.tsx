import { renderWithTheme } from 'helpers'

import Main from '.'

describe('<Main />', () => {
  it('should render the Main', () => {
    const { container } = renderWithTheme(<Main />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
