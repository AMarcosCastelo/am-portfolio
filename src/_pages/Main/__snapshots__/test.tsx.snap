// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the Main 1`] = `
.c4 {
  position: fixed;
  top: 2rem;
  left: 50%;
  -webkit-transform: translate(-50%,0);
  -ms-transform: translate(-50%,0);
  transform: translate(-50%,0);
  background-color: #FCF6F4;
  padding: 0.3rem;
  border-radius: 50%;
  border: 1px solid #000;
  width: 2.5rem;
  height: 2.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  z-index: 3;
  cursor: pointer;
}

.c4:hover {
  background-color: rgba(0,255,0,0.4);
  box-shadow: 0 0 8px 6px rgba(0,255,0,0.2);
}

.c4:hover > *:first-child {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c5 {
  display: inline-block;
  color: #000;
  font-family: 'Pacifico',cursive;
  position: fixed;
  left: 2rem;
  top: 2rem;
  z-index: 3;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: fixed;
  bottom: 0;
  left: 2rem;
  z-index: 3;
}

.c2 > *:not(:last-child) {
  margin: 0.5rem 0;
}

.c3 {
  width: 2px;
  height: 8em;
  background-color: #000;
}

.c0 {
  background: #FCF6F4;
  width: 100vw;
  height: 100vh;
  overflow: hidden;
  position: relative;
}

.c0 h2,
.c0 h3,
.c0 h4,
.c0 h5,
.c0 h6 {
  font-family: 'Karla',sans-serif;
  font-weight: 500;
}

.c1 {
  padding: 2rem;
}

.c8 {
  color: #000;
  position: absolute;
  top: 2rem;
  right: calc(1rem + 2vw);
  -webkit-text-decoration: none;
  text-decoration: none;
  z-index: 1;
}

.c9 {
  color: #000;
  position: absolute;
  top: 50%;
  right: calc(1rem + 2vw);
  -webkit-transform: rotate(90deg) translate(-50%,-50%);
  -ms-transform: rotate(90deg) translate(-50%,-50%);
  transform: rotate(90deg) translate(-50%,-50%);
  -webkit-text-decoration: none;
  text-decoration: none;
  z-index: 1;
}

.c10 {
  position: absolute;
  bottom: 1rem;
  left: 0;
  right: 0;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
}

.c11 {
  color: #000;
  -webkit-text-decoration: none;
  text-decoration: none;
  z-index: 1;
}

.c12 {
  color: #000;
  -webkit-text-decoration: none;
  text-decoration: none;
  z-index: 1;
}

.c7 {
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  border: none;
  outline: none;
  background-color: transparent;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: all 1s ease;
  transition: all 1s ease;
}

.c7 >:first-child {
  -webkit-animation: fEWCgj 1.5s linear infinite;
  animation: fEWCgj 1.5s linear infinite;
}

.c7 >:last-child {
  display: inline-block;
  padding-top: 1rem;
}

.c6 {
  position: absolute;
  background-color: #000;
  top: 0;
  bottom: 0;
  right: 50%;
  width: 0%;
  height: 0%;
  z-index: 1;
  -webkit-transition: height 0.5s ease,width 1s ease 0.5s;
  transition: height 0.5s ease,width 1s ease 0.5s;
}

<main
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div>
        <a
          href="/#"
        >
          <svg
            aria-hidden="true"
            class="prefix__svg-inline--fa prefix__fa-github prefix__fa-w-16"
            data-icon="github"
            data-prefix="fab"
            fill="#000"
            height="25"
            viewBox="0 0 496 512"
            width="25"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
              fill="#000"
            />
          </svg>
        </a>
      </div>
      <div>
        <a
          href="/#"
        >
          <svg
            aria-hidden="true"
            class="prefix__svg-inline--fa prefix__fa-facebook prefix__fa-w-16"
            data-icon="facebook"
            data-prefix="fab"
            fill="#000"
            height="25"
            viewBox="0 0 512 512"
            width="25"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M504 256C504 119 393 8 256 8S8 119 8 256c0 123.78 90.69 226.38 209.25 245V327.69h-63V256h63v-54.64c0-62.15 37-96.48 93.67-96.48 27.14 0 55.52 4.84 55.52 4.84v61h-31.28c-30.8 0-40.41 19.12-40.41 38.73V256h68.78l-11 71.69h-57.78V501C413.31 482.38 504 379.78 504 256z"
              fill="#000"
            />
          </svg>
        </a>
      </div>
      <div>
        <a
          href="/#"
        >
          <svg
            aria-hidden="true"
            class="prefix__svg-inline--fa prefix__fa-twitter prefix__fa-w-16"
            data-icon="twitter"
            data-prefix="fab"
            fill="#000"
            height="25"
            viewBox="0 0 512 512"
            width="25"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
              fill="#000"
            />
          </svg>
        </a>
      </div>
      <span
        class="c3"
      />
    </div>
    <button
      class="c4"
    >
      <a
        href="/"
      >
        <svg
          aria-hidden="true"
          class="prefix__svg-inline--fa prefix__fa-power-off prefix__fa-w-16"
          data-icon="power-off"
          data-prefix="fas"
          fill="currentColor"
          height="30"
          viewBox="0 0 512 512"
          width="30"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M400 54.1c63 45 104 118.6 104 201.9 0 136.8-110.8 247.7-247.5 248C120 504.3 8.2 393 8 256.4 7.9 173.1 48.9 99.3 111.8 54.2c11.7-8.3 28-4.8 35 7.7L162.6 90c5.9 10.5 3.1 23.8-6.6 31-41.5 30.8-68 79.6-68 134.9-.1 92.3 74.5 168.1 168 168.1 91.6 0 168.6-74.2 168-169.1-.3-51.8-24.7-101.8-68.1-134-9.7-7.2-12.4-20.5-6.5-30.9l15.8-28.1c7-12.4 23.2-16.1 34.8-7.8zM296 264V24c0-13.3-10.7-24-24-24h-32c-13.3 0-24 10.7-24 24v240c0 13.3 10.7 24 24 24h32c13.3 0 24-10.7 24-24z"
            fill="currentColor"
          />
        </svg>
      </a>
    </button>
    <h1
      class="c5"
    >
      AM
    </h1>
    <div
      class="c6"
    />
    <button
      class="c7"
    >
      <svg
        aria-hidden="true"
        class="prefix__svg-inline--fa prefix__fa-yin-yang prefix__fa-w-16"
        data-icon="yin-yang"
        data-prefix="fas"
        height="200"
        viewBox="0 0 496 512"
        width="200"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M248 8C111.03 8 0 119.03 0 256s111.03 248 248 248 248-111.03 248-248S384.97 8 248 8zm0 376c-17.67 0-32-14.33-32-32s14.33-32 32-32 32 14.33 32 32-14.33 32-32 32zm0-128c-53.02 0-96 42.98-96 96s42.98 96 96 96c-106.04 0-192-85.96-192-192S141.96 64 248 64c53.02 0 96 42.98 96 96s-42.98 96-96 96zm0-128c-17.67 0-32 14.33-32 32s14.33 32 32 32 32-14.33 32-32-14.33-32-32-32z"
        />
      </svg>
      <span>
        click here
      </span>
    </button>
    <a
      class="c8"
      href="mailto:marcos-12marcos@hotmail.com"
      target="_blank"
    >
      <h2>
        Say hi...
      </h2>
    </a>
    <a
      class="c9"
      href="/work"
    >
      <h2>
        Work
      </h2>
    </a>
    <div
      class="c10"
    >
      <a
        class="c11"
        href="/about"
      >
        <h2>
          About
        </h2>
      </a>
      <a
        class="c12"
        href="/mySkills"
      >
        <h2>
          My Skills
        </h2>
      </a>
    </div>
  </div>
</main>
`;
